aspnetcore/fundamentals/servers/httpsys.md

```diff
@@ -256,6 +256,45 @@ In Visual Studio, the default launch profile is for IIS Express. To run the proj
 
 For apps hosted by HTTP.sys that interact with requests from the Internet or a corporate network, additional configuration might be required when hosting behind proxy servers and load balancers. For more information, see [Configure ASP.NET Core to work with proxy servers and load balancers](xref:host-and-deploy/proxy-load-balancer).
 
+## Advanced HTTP/2 features to support gRPC
+
+Additional HTTP/2 features in HTTP.sys support gRPC, including support for response trailers and sending reset frames.
+
+Requirements to run gRPC with HTTP.SYS:
+
+* Windows 10 Version 2004, OS Build 20300.1000 or later
+* TLS 1.2 or later connection
+
+### Trailers
+
+HTTP Trailers are similar to HTTP Headers, except they are sent after the response body is sent. For HTTP.sys, only HTTP/2 response trailers are supported.
+
+```csharp
+if (httpContext.Response.SupportsTrailers())
+{
+    // Write body
+    httpContext.Response.WriteAsync("Hello world");
+
+    httpContext.Response.AppendTrailer("trailername", "TrailerValue");
+}
+```
+
+In the preceding example code:
+
+* `SupportsTrailers` ensures that trailers are supported for the response.
+* `AppendTrailer` appends the trailer.
```

Comment:
There's also DeclareTrailer to be called before starting the response body.

Reply:
Not for HTTP/2 trailers.

Reply:
? I thought it was still recommended to declare what trailers should be expected.

Reply:
Not that I've seen. No gRPC implementation does that with trailers.

Is that in a spec?

Reply:
Why is it recommend to declare trailers for HTTP/2?

Reply:
Yes the HTTP/1.1 spec discusses Trailer with chunked because that was the only format that supported trailers at the time. https://tools.ietf.org/html/rfc7230#section-4.4 

>    When a message includes a message body encoded with the chunked
   transfer coding and the sender desires to send metadata in the form
   of trailer fields at the end of the message, the sender SHOULD
   generate a Trailer header field before the message body to indicate
   which fields will be present in the trailers.  This allows the
   recipient to prepare for receipt of that metadata before it starts
   processing the body, which is useful if the message is being streamed
   and the recipient wishes to confirm an integrity check on the fly.

Http/2 changed the wire format but it didn't change the semantics of most headers. I don't see anything in the HTTP/2 spec that negates the recommendation to set a Trailer header, and it's still used in the HTTP/2 samples.

```
     HTTP/1.1 200 OK                  HEADERS
     Content-Type: image/jpeg   ==>     - END_STREAM
     Transfer-Encoding: chunked         + END_HEADERS
     Trailer: Foo                         :status = 200
                                          content-length = 123
     123                                  content-type = image/jpeg
     {binary data}                        trailer = Foo
     0
     Foo: bar                         DATA
                                        - END_STREAM
                                      {binary data}

                                      HEADERS
                                        + END_STREAM
                                        + END_HEADERS
                                          foo = bar
```

Reply:
Okay, bringing back declare trailers.

Reply:
Ok.

Do you still want to mention `trailer` even though HTTP/2 clients don't use it? We should at least say that it is optional, and generally not required by HTTP/2 clients.

Reply:
It was never required by HTTP/1.1 clients either, it was only a polite warning of things to come. I don't see how it's any different in HTTP/2.